index.js

```diff
@@ -519,10 +592,14 @@ var assemble = function () {
  */
 module.exports = function (options) {
 
-	// setup assembly
-	setup(options);
+	try {
+
+		// setup assembly
+		setup(options);
+
+		// assemble
+		assemble();
 
-	// assemble
-	assemble();
+	} catch(e) {}
 
```

Comment:
Good call. I simplified this a bit by eliminating the try catch in the `getMatter` function and just calling  `handleError` here. Only down side is the inability to get the file name for front-matter errors.